app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt

```diff
@@ -53,4 +112,39 @@ class PresetsFragment : BaseFragment() {
             }
         }
     }
+
+    inner class CategoriesPagerAdapter(fm: FragmentManager, private val categories: List<String>) :
+        FragmentStateAdapter(fm, viewLifecycleOwner.lifecycle) {
+
+        override fun getItemCount(): Int {
+            return ceil(categories.size / MAX_CATEGORIES.toDouble()).toInt()
+        }
+
+        override fun createFragment(position: Int): Fragment {
+            val startPosition = position * MAX_CATEGORIES
+            val subList = categories.subList(
+                startPosition,
+                min(categories.size, startPosition + MAX_CATEGORIES)
+            )
+
+            return CategoriesFragment.newInstance(subList)
+        }
+    }
+
+    inner class PhrasesPagerAdapter(fm: FragmentManager, private val phrases: List<String>) :
+        FragmentStateAdapter(fm, viewLifecycleOwner.lifecycle) {
+
+        override fun getItemCount(): Int {
+            return ceil(phrases.size / MAX_PHRASES.toDouble()).toInt()
+        }
+
+        override fun createFragment(position: Int): Fragment {
```

Comment:
we should have infinite scrolling eventually :]

Reply:
That's coming in the next PR!